kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,453 @@
+        return false;
+    }
+
+    @Override
+    public void prepareContainerUpdate(String id,
+                                       KieContainerInstance kieContainerInstance,
+                                       Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public void updateContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public void disposeContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public List<Object> getAppComponents(SupportedTransports type) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public <T> T getAppComponents(Class<T> serviceType) {
+        return null;
+    }
+
+    @Override
+    public String getImplementedCapability() {
+        return KieServerConstants.CAPABILITY_BPM_KAFKA;
+    }
+
+    @Override
+    public List<Object> getServices() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    @Override
+    public Integer getStartOrder() {
+        return 20;
+    }
+
+    @Override
+    public void run() {
+        Duration duration =
+                Duration.ofSeconds(Long.getLong(KAFKA_EXTENSION_PREFIX + "poll.interval", 10L));
+        logger.trace("Start polling kafka consumer every {} seconds", duration.getSeconds());
+        while (consumerReady.get()) {
+            if (isSubscribed()) {
+                ConsumerRecords<String, byte[]> events = ConsumerRecords.empty();
+                consumerLock.lock();
+                try {
+                    try {
+                        events = consumer.poll(duration);
+                    } catch (WakeupException ex) {
+                        logger.trace("Kafka wait interrupted");
+                    } catch (Exception ex) {
+                        logger.error("Error polling kafka consumer", ex);
+                    }
+                } finally {
+                    consumerLock.unlock();
+                }
+                if (consumerReady.get() && !events.isEmpty()) {
+                    if (logger.isDebugEnabled()) {
+                        Map<String, Integer> eventsPerTopic = new HashMap<>();
+                        for (ConsumerRecord<String, byte[]> event : events) {
+                            eventsPerTopic.compute(event.topic(), (k, v) -> v == null ? 1 : v++);
+                        }
+                        logger.debug("Number of events received per topic {}", eventsPerTopic);
+                    }
+                    for (ConsumerRecord<String, byte[]> event : events) {
+                        notifyService.submit(() -> processEvent(event));
+                    }
+                }
+            }
+        }
```

Comment:
if we have consumer but not topics this loop does not sleep. we will eat CPU non stop without interruption.

Reply:
good catch!! This bug was provoked by changing from ScheduledService to backgroung thread. Fixed used condition lock

Reply:
good catch, fixed with a condition object